src/tests/test-bpsys.cc

```diff
@@ -138,6 +138,11 @@ int open(const char *pathname, int flags)
     return nfd;
 }
 
+int open(const char *pathname, int flags, mode_t mode)
+{
+    return open(pathname, flags);
```

Comment:
This doesn't hand `mode` argument properly, for example if a file descriptor was not opened for write then writing should fail (and similarly if opened `O_WRONLY` then read should fail).

At least add a `TODO` comment.

Reply:
Added `ToDo`